service/src/main/java/bio/terra/javatemplate/iam/SamClient.java

```diff
@@ -27,12 +30,16 @@ private ApiClient getApiClient(String accessToken) {
   }
 
   private ApiClient getApiClient() {
-    var okHttpClientWithTracing =
-        this.okHttpClient
-            .newBuilder()
-            .addInterceptor(new OkHttpClientTracingInterceptor(Tracing.getTracer()))
-            .build();
-    return new ApiClient().setHttpClient(okHttpClientWithTracing).setBasePath(samConfig.basePath());
+    var client =
+        openTelemetry
+            .map(
+                otel ->
+                    this.okHttpClient
+                        .newBuilder()
+                        .addInterceptor(new OkHttpClientTracingInterceptor(otel))
+                        .build())
+            .orElse(this.okHttpClient);
```

Comment:
Why is this done here instead of in the constructor? Do we need different `okHttpClient`s for each caller of `getApiClient()`?